e2etests/tests/routes.go

```diff
@@ -0,0 +1,276 @@
+)
+
+var _ = Describe("Routes between bgp and the fabric", Ordered, func() {
+	var cs clientset.Interface
+	routerPods := []*corev1.Pod{}
+	frrk8sPods := []*corev1.Pod{}
+
+	vniRed := v1alpha1.VNI{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "red",
+			Namespace: openperouter.Namespace,
+		},
+		Spec: v1alpha1.VNISpec{
+			ASN:       64514,
+			VNI:       100,
+			LocalCIDR: "192.169.10.0/24",
+			HostASN:   ptr.To(uint32(64515)),
+		},
+	}
+
+	vniBlue := v1alpha1.VNI{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "blue",
+			Namespace: openperouter.Namespace,
+		},
+		Spec: v1alpha1.VNISpec{
+			ASN:       64514,
+			VNI:       200,
+			LocalCIDR: "192.169.11.0/24",
+			HostASN:   ptr.To(uint32(64515)),
+		},
+	}
+
+	frrK8sConfigRed, err := frrk8s.ConfigFromVNI(vniRed)
+	if err != nil {
+		panic(err)
+	}
+	frrK8sConfigBlue, err := frrk8s.ConfigFromVNI(vniBlue)
+	if err != nil {
+		panic(err)
+	}
+
+	changeLeafPrefixes := func(leaf infra.Leaf, redPrefixes, bluePrefixes []string) {
+		leafConfiguration := infra.LeafConfiguration{
+			Leaf: leaf,
+			Red: infra.Addresses{
+				IPV4: redPrefixes,
+			},
+			Blue: infra.Addresses{
+				IPV4: bluePrefixes,
+			},
+		}
+		config, err := infra.LeafConfigToFRR(leafConfiguration)
+		Expect(err).NotTo(HaveOccurred())
+		err = leaf.ReloadConfig(config)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	removeLeafPrefixes := func(leaf infra.Leaf) {
+		changeLeafPrefixes(leaf, []string{}, []string{})
+	}
+
+	BeforeAll(func() {
+		err := Updater.CleanAll()
+		Expect(err).NotTo(HaveOccurred())
+
+		cs = k8sclient.New()
+		routerPods, err = openperouter.RouterPods(cs)
+		Expect(err).NotTo(HaveOccurred())
+		frrk8sPods, err = frrk8s.Pods(cs)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = Updater.Update(config.Resources{
+			Underlays: []v1alpha1.Underlay{
+				infra.Underlay,
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterAll(func() {
+		err := Updater.CleanAll()
+		Expect(err).NotTo(HaveOccurred())
+		By("waiting for the router pod to rollout after removing the underlay")
+		Eventually(func() error {
+			return openperouter.DaemonsetRolled(cs, routerPods)
+		}, 2*time.Minute, time.Second).ShouldNot(HaveOccurred())
+
+		removeLeafPrefixes(infra.LeafAConfig)
+		removeLeafPrefixes(infra.LeafBConfig)
+	})
+
+	BeforeEach(func() {
+		err := Updater.CleanButUnderlay()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		err := Updater.CleanButUnderlay()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Context("with vnis", func() {
+		BeforeEach(func() {
+			err := Updater.Update(config.Resources{
+				VNIs: []v1alpha1.VNI{
+					vniRed,
+					vniBlue,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			removeLeafPrefixes(infra.LeafAConfig)
+			removeLeafPrefixes(infra.LeafBConfig)
+		})
+
+		It("receives type 5 routes from the fabric", func() {
+			Contains := true
+			checkRouteForLeaf := func(leaf infra.Leaf, vni v1alpha1.VNI, mustContain bool, prefixes []string) {
```

Comment:
can/should this be From instead of For?

Reply:
I don't mind changing it. It's either "check the routes for that leaf" or check the routes coming from that leaf.